src/util/guard.cpp

```diff
@@ -108,10 +106,10 @@ guardt &operator -= (guardt &g1, const guardt &g2)
       it2!=op2.end();
       ++it2)
   {
-    while(it1!=op1.end() && *it1<*it2)
-      ++it1;
     if(it1!=op1.end() && *it1==*it2)
       it1=op1.erase(it1);
+    else
```

Comment:
I don't think this works? For example, `a & b` - `b` would not erase the `b`, since `it1` now has no way of moving except when erasing an element.

Reply:
Relatedly, for vectors this has worst-case quadratic cost. How about making a utility inplace-set-difference that provides a good place to benchmark this sort of repeated single-element erase vs. `std::set_difference` followed by swap?

Reply:
I agree that this would not work with `a & b` - `b`,  but as the commit message says: we never construct those.  My proposal for a full solution is #310: use BDDs. I'm open to all other ideas. But right now it's just computational overhead...

Reply:
In that case I suggest renaming this `remove_prefix_conjuncts`, since its behaviour is much more specific than a generic `operator-=`, and add a doxy comment specifying that's all it can do. Also consider using `std::mismatch`?